ivy/functional/ivy/device.py

```diff
@@ -126,12 +128,26 @@ def __exit__(
         "cpu"
         """
         ivy.unset_default_device()
+        ivy.unset_soft_device_mode()
         if self and (exc_type is not None):
             print(exc_tb)
             raise exc_val
         return self
 
 
+def handle_soft_device_variable(*args, **kwargs):
+    default_device = ivy.default_device()
```

Comment:
This function expects native arrays and will return native arrays.
For this, You should set `array_mode` to false after entering this function then unset it before returning.